csrc/nv_internal/tensorrt_llm/kernels/cutlass_kernels/moe_gemm/moe_gemm_template_dispatch.h

```diff
@@ -728,10 +728,15 @@ void MoeGemmRunner<T, WeightType, OutputType, ScaleBiasType>::dispatchToArch(
       if (inputs.gemm_config.sm_version >= 90) {
         bool is_same_sm = inputs.gemm_config.sm_version == sm_;
         // gemm_config.sm_version indicates the kernel pipeline, which is always 100 for 100, 103,
-        // 110 below logging helps confirming the cutlass pipeline matches the device major version
+        // 110, 120, 121. Below checks confirm the cutlass pipeline matches the device major
+        // version.
         bool is_sm110 = inputs.gemm_config.sm_version == 100 && sm_ == 110;
         bool is_sm103 = inputs.gemm_config.sm_version == 100 && sm_ == 103;
-        TLLM_CHECK_WITH_INFO(is_same_sm || is_sm110 || is_sm103,
+        // SM120 and SM121 are architecturally identical, accept configs from either
+        bool is_sm120 =
+            (inputs.gemm_config.sm_version == 100 || inputs.gemm_config.sm_version == 120) &&
```

Comment:
`inputs.gemm_config.sm_version == 100` 

Why sm100's config is also applicable to sm120/121?

Reply:
updated as ``bool is_sm120 = (inputs.gemm_config.sm_version == 120) && (sm_ == 120 || sm_ == 121);``

Reply:
This looks more reasonable to me.